src/main/scala/org/scoverage/coveralls/CoverallPayloadWriter.scala

```diff
@@ -31,8 +31,16 @@ class CoverallPayloadWriter(
     def writeOpt(fieldName: String, holder: Option[String]) =
       holder foreach { gen.writeStringField(fieldName, _) }
 
-    writeOpt("repo_token", repoToken)
-    writeOpt("service_name", service.map(_.name))
+    coverallsAuth match {
+      case CoverallsRepoToken(token) =>
+        gen.writeStringField("repo_token", token)
```

Comment:
it appears that `service_name` was missed out in this case, which causes the upload job to fail for private repositories where the coveralls repo token is used:
>Uploading to https://coveralls.io/ failed: Provided service_job_id but not service_name.